src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditCategory.java

```diff
@@ -0,0 +1,44 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.impl;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+
+public enum AuditCategory {
+    BAD_HEADERS,
+    FAILED_LOGIN,
+    MISSING_PRIVILEGES,
+    GRANTED_PRIVILEGES,
+    OPENDISTRO_SECURITY_INDEX_ATTEMPT,
+    SSL_EXCEPTION,
+    AUTHENTICATED,
+    COMPLIANCE_DOC_READ,
+    COMPLIANCE_DOC_WRITE,
+    COMPLIANCE_EXTERNAL_CONFIG,
+    COMPLIANCE_INTERNAL_CONFIG_READ,
+    COMPLIANCE_INTERNAL_CONFIG_WRITE;
+
+    private static Logger log = LogManager.getLogger(AuditCategory.class);
+
+    public static EnumSet<AuditCategory> parse(final List<String> categories) {
```

Comment:
Consider using `Iterable` or `Collection` instead of a `List`. There is no need to know `size()`, so `Iterable` (preferred) or `Collection` should be sufficient.